src/pools/pool.service.spec.ts

```diff
@@ -0,0 +1,161 @@
+  const poolProvider = new PoolsStaticRepository(pools as SDKPool[]);
+  const poolsRepositories: BalancerDataRepositories = {
+    ...dataRepositories,
+    ...{
+      pools: poolProvider,
+    },
+  };
+  return poolsRepositories;
+}
+
+describe('Pool Service', () => {
+  describe('expandPool', () => {
+    let linearPools: Pool[];
+    let boostedPoolInfo: BoostedInfo;
+    let boostedPool: Pool;
+
+    beforeAll(() => {
+      const linearPoolDescriptions: LinearParams = {
+        pools: [
+          {
+            tokens: {
+              wrappedSymbol: 'aDAI',
+              mainSymbol: 'DAI',
+            },
+            balance: '1000000',
+          },
+          {
+            tokens: {
+              wrappedSymbol: 'aUSDC',
+              mainSymbol: 'USDC',
+            },
+            balance: '500000',
+          },
+          {
+            tokens: {
+              wrappedSymbol: 'aUSDT',
+              mainSymbol: 'USDT',
+            },
+            balance: '500000',
+          },
+        ]
+      };
+      
+      const linearPoolInfo = factories.linearPools.transient(linearPoolDescriptions).build();
+      linearPools = linearPoolInfo.linearPools;
+      
+      boostedPoolInfo = factories.boostedPool
+        .transient({
+          linearPoolInfo,
+          rootId: 'phantom_boosted_1',
+          rootAddress: 'address_phantom_boosted_1',
+        })
+        .build();
+      boostedPool = boostedPoolInfo.rootPool;
+    });
+
+    it('Should return a pool without any subpools as exactly the same', async () => {
+      const pool: Pool =  factories.poolBase.build();
+      const poolService = new PoolService(pool, networkConfig, getPoolsRepository([]));
+      await poolService.expandPool();
+      expect(poolService.pool).toEqual(pool);
+    });
+    
+    it('Should add immediate sub-pool information to a pool', async () => {
+      const poolService = new PoolService(boostedPool, networkConfig, getPoolsRepository([...linearPools, boostedPool]));
+      await poolService.expandPool();
+      const tokens = poolService.pool.tokens;
+      expect(tokens.length).toBe(4);
+      expect(tokens[0].token?.pool).toBeTruthy()
+      const tokenTreePoolItems = [
+        'id', 'address', 'poolType', 'totalShares', 'mainIndex'
+      ]
+      expect(tokens[0].token?.pool).toMatchObject(_.pick(linearPools[0], tokenTreePoolItems));
+      expect(tokens[1].token?.pool).toMatchObject(_.pick(linearPools[1], tokenTreePoolItems));
+      expect(tokens[2].token?.pool).toMatchObject(_.pick(linearPools[2], tokenTreePoolItems));
+      expect(tokens[3].token).toBeFalsy(); // This is the BPT token, it shouldn't add a subpool of itself
+    });
+
+    it('Should add multiple layers deep of sub-pool to the pool', async () => {
+      const stableLinearPoolDescription: LinearParams = {
+        pools: [
+          {
+            tokens: {
+              wrappedSymbol: 'aSTABLE',
+              mainSymbol: 'STABLE',
```

Comment:
We are no longer fetching linear pool specific data for the pool schema in the frontend. Not sure if something needs to be changed in the API to reflect that? Here is where I stripped out all the fetching of that data in the multicaller in the frontend: https://github.com/balancer-labs/frontend-v2/pull/2440/commits/07539448dee923e0246805f3c0dbec73141273f9

Reply:
This is just for generating test data. The API doesn't fetch any linear pool specific data.